x-pack/legacy/plugins/maps/common/descriptor_types.d.ts

```diff
@@ -18,3 +18,30 @@ export interface ILayerDescriptor {
   id: string;
   label?: string;
 }
+
+interface IFieldMetaDescriptor {
+  [key: string]: string | number | unknown;
+}
+
+interface IRangeFieldMetaDescriptor extends IFieldMetaDescriptor {
```

Comment:
`IRangeFieldMetaDescriptor` is not a descriptive enough name. There are two objects that could be called `IRangeFieldMetaDescriptor`.
1) fieldMetaFromFeatures
2) fieldMeta from elasticsearch aggregation call.

This instance is the first, field meta from features. How about renaming to `IRangeFieldMetaFromFeatures`. I think you can drop the `Descriptor` part here since there is not fieldMeta object that this could be confused with